sql/analyzer/indexed_joins.go

```diff
@@ -300,6 +300,19 @@ func addLookupJoins(m *memo.Memo) error {
 					Nullmask: nullmask,
 				},
 			}
+			var filters []memo.ScalarExpr
+			for _, filter := range rel.Filter {
+				found := false
+				for _, matchedFilter := range matchedFilters {
+					if filter == matchedFilter {
```

Comment:
you might be able to just match with `e.ExprId()`

Reply:
Good catch! That's not only more efficient, but that allows us to detect matches through table aliases.

Reply:
Switching to use e.ExprId() appears to have broken some correctness tests. I'm going to merge the previous version.